### PR TITLE
Upgrade to Next.js 14 and latest i18next

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-root
app-content-pages
app-project

## Describe your changes
- Upgrade Next and the i18next packages in all apps and libs

## How to Review

- I tested this by running each package's Storybook where available, and built + ran each app locally to review the checklist below, plus basic functionality of user groups pages.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out

## Maintenance
- [ ] If not from dependabot, the PR creator has described the update (major, minor, or patch version, changelog)

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Upgrading to superagent v9 should fix that build error.
https://github.com/ladjs/superagent/issues/1786

See also https://github.com/zooniverse/Panoptes-Front-End/issues/7143

**Comment by goplayoutside3** [comment]
Out of curiosity, why is this a build problem from app-root, but not for the other Next.js apps in the monorepo? All use superagent in PJC and the new lib-panoptes-js client. Because app-root is an App Router?

**Comment by eatyourgreens** [comment]
I’ve been wondering the same thing. The error’s coming from the publications page, when it requests the project avatars in Node, but that works fine with the pages router. 

The error itself is caused by the webpack loader in Next.js. It’s trying to use the CJS loader with an ESM module.

**Comment by eatyourgreens** [comment]
Oh, and the package with the broken export, `hexoid`, hasn’t been touched in 4 years. 
https://github.com/lukeed/hexoid/issues/7

**Comment by goplayoutside3** [comment]
> The error’s coming from the publications page, when it requests the project avatars in Node, but that works fine with the pages router.

If I completely delete the `/publications` route from the app router, the error becomes `TypeError: a is not a function` for every route. So it must just encounter data fetching on the publications page first...

But then if I remove `@zooniverse/react-components` from `optimizePackageImports` in next.config.mjs, the build errors become complaints about using React hooks in server components.

The errors are super confusing in general 🤷‍♀️

File: packages/app-root/next.config.mjs [hunk 1]
```diff
@@ -11,11 +11,15 @@ const nextConfig = {
...
-    ],
+    ]
   },
   webpack: (config, options) => {
     config.resolve = {
       ...config.resolve,
+      alias: {
+        ...config.resolve.alias,
+        hexoid: 'hexoid/dist/index.js'
+      },
```
Comment (goplayoutside3) [inline packages/app-root/next.config.mjs:22]: This punts the issue of upgrading Superagent, but does fix the build errors 🤔

**Comment by eatyourgreens** [comment]
I think there might be a superagent request to get the auth'ed user on every route, but I'm not sure. 

I tried forcing superagent to v10, in `package.json`, but the app router build still errors. 🫤

File: packages/app-content-pages/src/api/team/processTeamData.mock.json [hunk 2]
```diff
@@ -64,7 +64,7 @@
           "fields": {
             "title": "headshot",
             "file": {
-              "url": "https://placekitten.com/300/300",
```
Comment (goplayoutside3) [inline packages/app-content-pages/src/api/team/processTeamData.mock.json:67]: placekitten works inconsistently, so I replaced with our simple-avatar image.

File: packages/app-content-pages/src/screens/Publications/Publications.spec.js [hunk 3]
```diff
@@ -36,11 +36,6 @@ describe('Component > Publications Page', function () {
     )
   })
 
-  it('should have sidebar nav label', function () {
```
Comment (goplayoutside3) [inline packages/app-content-pages/src/screens/Publications/Publications.spec.js:39]: I'm not totally sure why these tests fail, but this app is going to be removed in a few weeks, so okay to delete for now.

File: packages/app-root/server/server.js [hunk 4]
```diff
@@ -8,7 +8,6 @@ const APP_ENV = process.env.APP_ENV || 'development'
 
 const hostnames = {
   development: 'local.zooniverse.org',
-  branch: 'fe-project-branch.preview.zooniverse.org',
```
Comment (goplayoutside3) [inline packages/app-root/server/server.js:11]: This is specifically used for app-project, not app-root.

File: packages/app-root/next.config.mjs (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-root/next.config.mjs:22]: I'm really curious as to why the pages router doesn't have a problem with this. We run some stuff on 14.2, and the upgrade from 13.5 was straightforward, but we're using the pages router.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

packages: app-root, app-project, app-content-pages

This PR is a major bump to our Next.JS dependency. Code changes are very simple, but since this is a major dependency for FEM, my main concern is making sure that Next.JS 14 doesn't break anything fundamental in our apps.

### Testing

Tested on localhost with macOS + Chrome. For each package, I ran `yarn build ; yarn start` instead of `yarn dev`.

- app-root:
  - I can load the [home page](https://local.zooniverse.org:3000/), sign in/sign out, and view the signed-in/signed-out pages respectively.
    - Minor quirk: signed-out page seems a bit slow, but that may be my computer having performance issues running videos.
  - I can view my [stats page](https://local.zooniverse.org:3000/users/darkeshard/stats).
- app-project:
  - I can load a [test project](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2022), and sign in/sign out
  - I can make and submit classifications on a [simple workflow](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2022/classify/workflow/22164).
- app-content-pages:
  - I can view the [Team](http://localhost:3000/about/team) page, and the [Publications](http://localhost:3000/about/publications) page

### Super Minor Dev Notes

- Thumbs up on replacing the placekitten placeholder images with our own hosted files. Side note, we probably should replace any unnecessary third-party assets/placeholders like this if they appear anywhere else in our codebase, to improve testing reliability. 🐱 

### Status

As far as I can tell, everything is looking good with this upgrade. I'm not seeing any obvious build issues, or obvious problems with the code changes. And most importantly, a standard manual test of all three packages indicate that each app functions as intended.

LGTM! 👍

**Pull request merged**